packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -182,88 +180,105 @@ library LibVerifying {
     )
         private
     {
-        TaikoData.Auction memory auction =
-            state.auctions[LibAuction.blockIdToBatchId(config, blk.blockId)];
-        // Keep track of the average.. and use it as a param in proof window
-        uint64 proofTime;
-
-        unchecked {
-            proofTime = uint64(
-                uint256(fc.provenAt - blk.proposedAt).max(
-                    uint256(auction.startedAt + config.auctionWindowInSec)
-                )
-            );
+        TaikoData.Auction memory auction;
+        {
+            uint64 batchId = LibAuction.batchForBlock(config, blk.blockId);
+            auction = state.auctions[batchId % config.auctionRingBufferSize];
+
+            // this may be false for system prover
+            if (auction.batchId == batchId) {
```

Comment:
Changed from 

`if (auction.batchId == batchId) {` to `if (auction.batchId != batchId) {`